visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -22,23 +69,45 @@ export class FileValidator {
 		return true
 	}
 
-	public static validate(file: { nodes: CodeMapNode[] }): Array<{ message: string; dataPath: string }> {
-		if (!file) {
-			return [{ message: "file is empty or invalid", dataPath: "empty or invalid file" }]
+	private static checkApiVersion(file: { nodes: CodeMapNode[] }) {
+		let apiVersion = this.getCodeChartaApiVersion()
+		let fileApiVersion = this.getFileApiVersion(file)
+
+		let apiVersionMajorMinor = [false, false]
+
+		let apiVersionMajor = apiVersion.split(".")[0]
+		let apiVersionMinor = apiVersion.split(".")[1]
+
+		let majorVersion = fileApiVersion.split(".")[0]
+		let minorVersion = fileApiVersion.split(".")[1]
+
+		if (majorVersion > apiVersionMajor) {
+			apiVersionMajorMinor[0] = true
+		} else if (minorVersion > apiVersionMinor) {
+			apiVersionMajorMinor[1] = true
 		}
-		let ajv = require("ajv")()
-		let compare = ajv.compile(require("./schema.json"))
-		let valid = compare(file)
+		return apiVersionMajorMinor
+	}
```

Comment:
This function is quite hard to read. How about splitting this (and the functions below) into multiple smaller separated pieces like:
```
	private isValidApiVersion(file: ExportCCFile): boolean {
		const hasApiVersion = file.apiVersion !== undefined
		const versionRegExp = new RegExp("[0-9]\.[0-9]")
		const isValidVersion = versionRegExp.test(file.apiVersion)
		return hasApiVersion && isValidVersion
	}

	private static fileHasHigherMajorVersion(file: ExportCCFile): boolean {
		let fileMajorVersion = this.getMajorVersion(file.apiVersion)
		let latestMajorVersion = this.getMajorVersion(latestApiVersion)

		return fileMajorVersion > latestMajorVersion
	}

	private static fileHasHigherMinorVersion(file: ExportCCFile): boolean {
		let fileMinorVersion = this.getMinorVersion(file.apiVersion)
		let latestMinorVersion = this.getMinorVersion(latestApiVersion)

		return fileMinorVersion > latestMinorVersion
	}

	private static getMajorVersion(version: string) {
		return version.split(".")[0]
	}

	private static getMinorVersion(version: string) {
		return version.split(".")[1]
	}
```

- The function `isValidApiVersion()` didn't really exist, but I think it makes sense, when you wanna access it to first check if its even valid. If not you should straight show an error message.
- The variable `latestApiVersion` is imported
  ```
  const latestApiVersion = require("../../../package.json").codecharta.apiVersion
  ```